brpc build failed in x64-linux

**Describe the bug (描述bug)**
I'm a member of [vcpkg](https://github.com/Microsoft/vcpkg) team, when I tried to update the version of `brpc` to 1.10.0, I got an error as below:
```
vcpkg/buildtrees/brpc/src/1.10.0-6977522ae8.clean/src/brpc/serialized_response.h:56:9: error: ‘int brpc::SerializedResponse::GetCachedSize() const’ marked ‘override’, but does not override
   56 |     int GetCachedSize() const override { return (int)_serialized.size(); }
      |         ^~~~~~~~~~~~~
vcpkg/buildtrees/brpc/src/1.10.0-6977522ae8.clean/src/brpc/serialized_response.h:74:10: error: ‘void brpc::SerializedResponse::SetCachedSize(int) const’ marked ‘override’, but does not override
   74 |     void SetCachedSize(int size) const override;
      |          ^~~~~~~~~~~~~
```
I found above issue occurs due to the version of `protobuf` in vcpkg was `4.25.1`, which there is no corresponding virtual function in the [base class](https://github.com/protocolbuffers/protobuf/blob/v4.25.1/src/google/protobuf/message.h).

I know from the [brcp documentation](https://github.com/apache/brpc/blob/master/docs/cn/getting_started.md#protobuf-30-325) that you are using protobuf version 3.0-3.25, so I'm not sure if you support version 4.25.1. 
I am currently trying to add a patch in vcpkg to remove the `override` keyword from those two functions, and I would like to confirm with you if this is a feasible solution.

**To Reproduce (复现方法)**
Build `brpc` 1.10.0 with `protobuf` 4.25.1.

**Expected behavior (期望行为)**
No error.

**Versions (各种版本)**
OS: Ubuntu 23.10
Compiler: GNU 13.2.0
brpc: 1.10.0
protobuf: 4.25.1

**Additional context/screenshots (更多上下文/截图)**

--- a/src/brpc/serialized_response.h
+++ b/src/brpc/serialized_response.h
@@ -53,7 +53,7 @@ public:
     void Clear() override;
     bool IsInitialized() const override;
     int ByteSize() const;
-    int GetCachedSize() const override { return (int)_serialized.size(); }
+    int GetCachedSize() const PB_422_OVERRIDE { return (int)_serialized.size(); }
     butil::IOBuf& serialized_data() { return _serialized; }
     const butil::IOBuf& serialized_data() const { return _serialized; }
 
@@ -71,7 +71,7 @@ private:
     void MergeFrom(const SerializedResponse& from);
     void SharedCtor();
     void SharedDtor();
-    void SetCachedSize(int size) const override;
+    void SetCachedSize(int size) const PB_422_OVERRIDE;
   
 private:
     butil::IOBuf _serialized;

